src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -560,6 +560,27 @@ private static void ConfigureAuthentication(WebApplicationBuilder builder, Dashb
                 {
                     OnCertificateValidated = context =>
                     {
+                        var options = context.HttpContext.RequestServices.GetRequiredService<IOptions<DashboardOptions>>().Value;
+                        if (options.Otlp.AllowedCertificates is { Count: > 0 } allowList)
```

Comment:
For my knowledge, why do we need our own custom option? Doesn't Kestrel have an existing setting for which certs are valid?

Reply:
ASP.NET Core cert auth provides settings for allowing or rejecting invalid certs, but there is just an event for determining whether a valid cert is allowed or not.

https://learn.microsoft.com/en-us/aspnet/core/security/authentication/certauth?view=aspnetcore-8.0 - see `OnCertificateValidated`

> Conceptually, the validation of the certificate is an authorization concern. Adding a check on, for example, an issuer or thumbprint in an authorization policy, rather than inside OnCertificateValidated, is perfectly acceptable.

This PR adds configuration and code for that check.